### PR TITLE
Enable XML report generation for FindBugs on the pull request builder

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] Reconfigure pom.xml
- [x] Enable FindBugs on https://jenkins.ci.cloudbees.com/job/plugins/job/ssh-agent-plugin/
- [x] Fix important issues

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
The code is clean => nothing to check

File: pom.xml [hunk 1]
```diff
@@ -186,10 +193,12 @@
       <plugin>
         <groupId>org.codehaus.mojo</groupId>
         <artifactId>findbugs-maven-plugin</artifactId>
-        <version>3.0.1</version>
+        <version>${findbugs-maven-plugin.version}</version>
         <configuration>
           <excludeFilterFile>src/findbugs/excludesFilter.xml</excludeFilterFile>
-          <failOnError>${maven.findbugs.failure.strict}</failOnError>
+          <failOnError>${findbugs.failOnError}</failOnError>
```
Comment (KostyaSha) [inline pom.xml]: If you set env variable, why do you need set it in configuration?

File: pom.xml [hunk 2]
```diff
@@ -186,10 +193,12 @@
...
+          <xmlOutput>true</xmlOutput>
+          <findbugsXmlWithMessages>true</findbugsXmlWithMessages>
```
Comment (KostyaSha) [inline pom.xml]: not used parameter

File: pom.xml (hunk 1 above)
Comment (KostyaSha) [inline pom.xml]: @stephenc should know, maybe i'm wrong.

Comment (oleg-nenashev) [inline pom.xml]: At least it highlights such variable for developers

File: pom.xml (hunk 2 above)
Comment (oleg-nenashev) [inline pom.xml]: Yes, makes sense to cleanup it

File: pom.xml (hunk 1 above)
Comment (stephenc) [inline pom.xml]: Get rid of it... you are just bloating the pom

File: pom.xml [hunk 3]
```diff
@@ -68,7 +68,8 @@
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
     <project.build.outputEncoding>UTF-8</project.build.outputEncoding>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <maven.findbugs.failure.strict>true</maven.findbugs.failure.strict>
+    <findbugs-maven-plugin.version>3.0.1</findbugs-maven-plugin.version>
+    <findbugs.failOnError>true</findbugs.failOnError>
```
Comment (stephenc) [inline pom.xml]: get rid of this

**Pull request merged**